local-tests/setup/shiva-client.ts

```diff
@@ -102,100 +110,82 @@ export class TestnetClient {
       const pollRes = await fetch(
         this._processEnvs.TESTNET_MANAGER_URL + '/test/poll/testnet/' + this._id
       );
-      const res = await pollRes.json();
+      const res: TestNetResponse<TestNetState> = await pollRes.json();
+      if (pollRes.status != 200) {
+        throw new ShivaError(res); // throw the response as an error
+      }
       state = res.body;
       console.log('found state to be', res);
-      if (state != 'Active' && state != 'UNKNOWN') {
-        await new Promise<void>((res, _) => {
-          setTimeout(() => {
-            res();
-          }, 500);
-        });
-      } else {
-        break;
-      }
+
+      await new Promise<void>((res, _) => {
+        setTimeout(() => {
+          res();
+        }, 500);
+      });
     }
 
     return state;
   }
 
   /**
-   * returns the config for a given testnet
-   * struct reference for config
+   * Returns the config for a given testnet
    */
-  public getTestnetConfig() {
-    return fetch(
+  public async getTestnetConfig() {
+    const res = await fetch(
       this._processEnvs.TESTNET_MANAGER_URL +
         '/test/get/info/testnet/' +
         this._id
-    )
-      .then((res: Response) => {
-        return res.json();
-      })
-      .then((info: TestNetResponse<TestNetInfo>) => {
-        this._info = info.body;
-        this._currentState = info.lastStateObserved as TestNetState;
-        console.log('setting testnet info: ', this._info);
-      });
+    );
+
+    let testnetConfigRes = _processTestnetResponse<boolean>(res);
+
+    return testnetConfigRes;
   }
 
   /**
    * Will wait for the NEXT epoch and return a resposne when the epoch has fully transitioned.
    * The return time is directly proportional to the epoch transition time config and where the network is with the current epoch.
    */
-  public transitionEpochAndWait() {
-    return fetch(
+  public async transitionEpochAndWait() {
+    const res = await fetch(
       this._processEnvs.TESTNET_MANAGER_URL +
         '/test/action/transition/epoch/wait/' +
         this._id
-    )
-      .then((res: Response) => {
-        if (res.status === 200) {
-          return res.json();
-        }
-      })
-      .then((body: any) => {
-        console.log('Stopped random peer: ', body);
-      });
+    );
+
+    let transitionEpochAndWaitRes = _processTestnetResponse<boolean>(res);
```

Comment:
Same remarks as the comment on `stopTestnet()` -- would be clearer to either `await` the result if you want to be able to inspect it on L158, or just `return` the function call.